code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -156,133 +158,36 @@
 		connected_ai.connected_robots -= src
 	qdel(wires)
 	qdel(module)
+	qdel(eye_lights)
 	wires = null
 	module = null
+	eye_lights = null
 	camera = null
 	cell = null
 	return ..()
 
 
 /mob/living/silicon/robot/proc/pick_module()
-	if(module)
+	if(module.type != /obj/item/weapon/robot_module)
 		return
 
-	var/list/modulelist = list("Standard", "Engineering", "Medical", "Miner", "Janitor", "Service")
+	var/list/modulelist = list("Standard" = /obj/item/weapon/robot_module/standard, \
+	"Engineering" = /obj/item/weapon/robot_module/engineering, \
+	"Medical" = /obj/item/weapon/robot_module/medical, \
+	"Miner" = /obj/item/weapon/robot_module/miner, \
+	"Janitor" = /obj/item/weapon/robot_module/janitor, \
+	"Service" = /obj/item/weapon/robot_module/butler)
```

Comment:
There is an argument to be made that keeping them makes it clearer that it's a list, but I guess it can go either way.